blockchain/types/src/avr_history.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Types for reading and writing historical Attestation Verification
+//! Reports from MobileCoin consensus enclaves
+
+use alloc::{string::String, vec::Vec};
+use hex::ToHex;
+use mc_attest_core::VerificationReport;
+use mc_crypto_keys::Ed25519Public;
+use mc_common::HashMap;
+use toml::{map::Map, value::Value};
+use url::Url;
+
+/// Object representing AVR history for all consensus nodes
+pub struct AVRHistoricalRecords {
+    /// AVR Records
+    pub avr_records: HashMap<Url, Vec<AVRHistoricalRecord>>,
+}
+
+impl AVRHistoricalRecords {
+    /// Create new records object
+    pub fn new(avr_records: HashMap<Url, Vec<AVRHistoricalRecord>>) -> Self {
+        Self { avr_records }
+    }
+
+    /// Get a toml representation of the AVR historical records
+    pub fn export_toml_string(&self) -> Result<String, toml::ser::Error>{
+        toml::to_string(&Map::from(self))
+    }
+}
+
+/// Stores a single historical AVR for each tuple of responder_id
+/// & block signing key  in the format specified in draft MCIP #43
+pub struct AVRHistoricalRecord {
+    /// Uri of the consensus node
+    pub node_url: Url,
```

Comment:
This should be a `ResponderId`, not a `Url`
```suggestion
    pub responder_id: ResponderId,
```